crates/pathfinder/src/state/sync.rs

```diff
@@ -811,6 +812,37 @@ async fn l2_update(
             .insert_block_header(&header)
             .context("Inserting block header into database")?;
 
+        let state_update_counts = StateUpdateCounts {
+            storage_diffs: state_update
+                .contract_updates
+                .iter()
+                .map(|(_, update)| update.storage.len())
+                .sum::<usize>()
+                .try_into()
+                .expect("ptr size is 64 bits"),
+            nonce_updates: state_update
+                .contract_updates
+                .iter()
+                .filter(|(_, update)| update.nonce.is_some())
+                .count()
+                .try_into()
+                .expect("ptr size is 64 bits"),
+            declared_classes: (state_update.declared_cairo_classes.len()
+                + state_update.declared_sierra_classes.len())
+            .try_into()
+            .expect("ptr size is 64 bits"),
+            deployed_contracts: state_update
+                .contract_updates
+                .iter()
+                .filter(|(_, update)| update.class.is_some())
+                .count()
+                .try_into()
+                .expect("ptr size is 64 bits"),
+        };
```

Comment:
Consider moving this as a method `StateUpdate::update_counts` - I know its likely only used in a single place right now; but its a lot of LoC in the middle here.

Reply:
Done